main/webapp/modules/core/scripts/project/history-panel.js

```diff
@@ -165,6 +170,22 @@ HistoryPanel.prototype._render = function() {
   this.resize();
 };
 
+HistoryPanel.prototype.showHideHistorySplash = function(clicked) {
+  var hideHistorySplash = Refine.getPreference("ui.history.hideHistorySplash", false);
+  var checkbox = $('#history-panel-checkbox');
+
+  if(checkbox.prop('checked')) {
+      Refine.setPreference('ui.history.hideHistorySplash', true);
+   }
```

Comment:
There is something weird here. You should just read the preference with the `Refine.getPreference()` once, at the construction time. After that, the DOM knows the value. Then, if the user changes the setting, then you write the preference, but still, you never have to read it back. It still current in the DOM.

I understand you code kinda works, but it's inefficient and openes the door to problems while complicating the logic, therefor the reading by others. Can you try to remodel it so it only calls `getPrefs()` once?

I haven't completely tested it, but the revision I got was not working on Chrome. Might be me, but... I'll check it back tomorrow.

Regards, Antoine

Reply:
> You should just read the preference with the Refine.getPreference() once, at the construction time. After that, the DOM knows the value. Then, if the user changes the setting, then you write the preference, but still, you never have to read it back. It still current in the DOM.

@antoine: Sir I just need some clarification. 

In this function, `getPreference` is called just once. Do you mean that `Refine.getPreference()` should be called once for both this function and that for the history splash?

Reply:
> > You should just read the preference with the Refine.getPreference() once, at the construction time. After that, the DOM knows the value. Then, if the user changes the setting, then you write the preference, but still, you never have to read it back. It still current in the DOM.
> 
> @antoine: Sir I just need some clarification.
> 
> In this function, `getPreference` is called just once. Do you mean that `Refine.getPreference()` should be called once for both this function and that for the history splash?

What I mean is that `getPreference()` is in a function that get called every time that a click is made, plus the fist time. So that means the actually call the `getPreference()` as many time as there are clicks, plus once for the initialisation. It should only be called ONCE per page loading. You can't have 2 authorities for the source of that, and once the page is loaded, the DOM owns the facts that the splash is displayed or visible. If that value changed behind the scene, there is no purpose to reflect it in the interface. At that stage, the user controls the interface and the display of the splashes.

And no, I see no need/advantage to manage Facet & History splash in the same code. Some function can be written in a generic way to be used by both splashes (or even more places), but then, it's because they are generic.

Regards, Antoine

Reply:
@antoine2711 : Please give me a second to check this